microshift.sh

```diff
@@ -0,0 +1,128 @@
+#!/bin/bash
+
+set -exuo pipefail
+
+export LC_ALL=C.UTF-8
+export LANG=C.UTF-8
+
+source tools.sh
+source snc-library.sh
+
+BUNDLE_TYPE="microshift"
+INSTALL_DIR=crc-tmp-install-data
+CRC_VM_NAME=${CRC_VM_NAME:-crc}
+BASE_DOMAIN=${CRC_BASE_DOMAIN:-testing}
+MIRROR=${MIRROR:-https://mirror.openshift.com/pub/openshift-v4/$ARCH/clients/ocp}
+OPENSHIFT_RELEASE_VERSION=4.12.0
+MICROSHIFT_NVR=${MICROSHIFT_NVR:-}
+
+if [ ! -n "${MICROSHIFT_NVR}" ]; then
+    echo "Please specify MICROSHIFT_NVR, which you can get from internal brew (ex. microshift-4.12.5-202302222013.p0.g0165ac4.assembly.4.12.5.el8 )"
+    exit 1
+fi
+
+if ! grep -q -i "release 8" /etc/redhat-release
+then
+  echo "This script only works for RHEL-8"
+  exit 1
+fi
+
+echo "Check if system is registered"
```

Comment:
Aren't these prefligth checks as well? (subscription and rhel8 system)

Reply:
Well, these could still be a `microshift_preflight` function instead of cluttering the main code path.